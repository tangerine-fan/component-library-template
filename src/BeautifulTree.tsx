export type BeautifulTreeProps = {
    id: string
    width: number
    height: number
  }
  
  export function BeautifulTree({ id, width, height }: BeautifulTreeProps): JSX.Element {
    return (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        id={id}
        viewBox={`0 0 ${width} ${height}`}
        style={{
          width: `${width}px`,
          height: `${height}px`,
        }}
        className={'beautiful-tree-react'}
      >
      {/* replace with real code to draw a tree */}
      <circle
        cx={width * 0.5}
        cy={height * 0.5}
        r={Math.min(width, height) * 0.5}
      />
      </svg>
    )
  }