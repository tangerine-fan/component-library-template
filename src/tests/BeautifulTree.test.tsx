import { BeautifulTree } from '../BeautifulTree'
import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'

describe('BeautifulTree', () => {
    it('renders a boring circle', () => {
        const rendered = render(
            <BeautifulTree
                id="my-boring-test-tree"
                width={200}
                height={200}
            />
        )

        // NOTE: Snapshots are like black boxes, and depending too much on them for
        //       testing can make our tests brittle. Use them sparingly.
        //
        // The first time that we run this test, it will generate a snapshot that
        // will be used to compare against future runs. We trust that the first
        // execution is correct because we saw it in Storybook before.
        expect(rendered).toMatchSnapshot()
    })
})