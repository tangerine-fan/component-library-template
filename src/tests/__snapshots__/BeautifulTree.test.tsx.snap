// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BeautifulTree > renders a boring circle 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="beautiful-tree-react"
        id="my-boring-test-tree"
        style="width: 200px; height: 200px;"
        viewBox="0 0 200 200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="100"
          cy="100"
          r="100"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="beautiful-tree-react"
      id="my-boring-test-tree"
      style="width: 200px; height: 200px;"
      viewBox="0 0 200 200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="100"
        cy="100"
        r="100"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
